File: software/examples/perl/example_hello_world.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletEPaper296x128;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your E-Paper 296x128 Bricklet

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $ep = Tinkerforge::BrickletEPaper296x128->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Use black background
$ep->fill_display($ep->COLOR_BLACK);

# Write big white "Hello World" in the middle of the screen
$ep->draw_text(16, 48, $ep->FONT_24X32, $ep->COLOR_WHITE, $ep->ORIENTATION_HORIZONTAL,
               "Hello World");
$ep->draw();

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_hello_world.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_e_paper_296x128'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your E-Paper 296x128 Bricklet

ipcon = IPConnection.new # Create IP connection
ep = BrickletEPaper296x128.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Use black background
ep.fill_display BrickletEPaper296x128::COLOR_BLACK

# Write big white "Hello World" in the middle of the screen
ep.draw_text 16, 48, BrickletEPaper296x128::FONT_24X32, \
             BrickletEPaper296x128::COLOR_WHITE, \
             BrickletEPaper296x128::ORIENTATION_HORIZONTAL, 'Hello World'
ep.draw

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
